test/e2e/scheduling/priorities.go

```diff
@@ -252,6 +252,21 @@ var _ = SIGDescribe("SchedulerPriorities [Serial]", func() {
 			if err := e2erc.DeleteRCAndWaitForGC(f.ClientSet, f.Namespace.Name, rc.Name); err != nil {
 				framework.Logf("Failed to cleanup replication controller %v: %v.", rc.Name, err)
 			}
+			// Just in case DeleteRCAndWaitForGC returns non-nil error but the pods are in
+			// terminating state, give them more time to get deleted so subsequent tests
+			// (e.g. [sig-scheduling] SchedulerPredicates [Serial] validates resource limits of pods that are allowed to run)
+			// do not process pods that are disappearing under their hands.
+			wait.PollImmediate(time.Second, 2*time.Minute, func() (bool, error) {
```

Comment:
Can you look into fixing `DeleteResourceAndWaitForGC` instead? According to its comment, it should already wait for it. Are you seeing `Failed to cleanup replication controller` in the logs? If so, we should fix those.

Reply:
Unfortunately, I have logs only from failed ci runs. `[sig-scheduling] SchedulerPredicates [Serial] validates resource limits of pods that are allowed to run` is green when `[sig-scheduling] SchedulerPredicates [Serial] validates resource limits of pods that are allowed to run` flakes. Also, it's reproducible only in ~5% of cases.

Reply:
In that case, I'm not convinced that this patch will solve the flakiness, as we don't understand the root cause of the leaking pods.

Reply:
Could you share a failed run?

Reply:
> Could you share a failed run?

In case of `[sig-scheduling] SchedulerPriorities [Serial] Pod should avoid nodes that have avoidPod annotation`, it will never fail because of pods not getting properly deleted. 

In case of `[sig-scheduling] SchedulerPredicates [Serial] validates resource limits of pods that are allowed to run`, just reported https://github.com/kubernetes/kubernetes/issues/94715

Reply:
@alculquicondor what about to fail when `e2erc.DeleteRCAndWaitForGC` returns an error instead? It's not a failure related to the test itself. Though, the test should still make sure all the resources created by it are cleaned.

Reply:
What we need to know is why it fails and fix that, because that's the function that is intended to do the cleanup. By waiting again after that function, we are duplicating code and moving the issue somewhere else.

Reply:
@alculquicondor I still have not been able to reproduce the case and get the error message when `e2erc.DeleteRCAndWaitForGC` is not deleting all the pods. Instead, I am updating `validates resource limits of pods that are allowed to run` to ignore all pods with `pod.DeletionTimestamp` set. Since those pods can disappear at any moment and make the calculated total resources unusable for decision making.